cvmfs/catalog.h

```diff
@@ -231,6 +231,7 @@ class Catalog : public SingleCopy {
   Counters& GetCounters() { return counters_; }
 
   inline const CatalogDatabase &database() const { return *database_; }
+  inline       CatalogDatabase &database()       { return *database_; }
```

Comment:
Please let's avoid non-const references in general.  Otherwise, when you read the code in a derived class, you don't know if what comes out of `database()` is modified or not.